runtime/POSIX/fd.c

```diff
@@ -38,6 +38,16 @@ static exe_disk_file_t *__get_sym_file(const char *pathname) {
   if (!pathname)
     return NULL;
 
+  //Handle the case where symbolic file is given as an absolute path, ie. /current/work/dir/A
+  if(pathname[0] == '/') {
+      char cwd[1024];
+      memset(cwd, 0, 1024);
+      getcwd(cwd, 1024);
```

Comment:
You should check the return value of `getcwd` - it might break for long paths when your buffer is too short.